### PR TITLE
AV-209807 : AKO does not honour readines probe for pods when Antrea with NPL is used

### PR DESCRIPTION
**Pull request by DixitAakash**
(#1455)

* AV-209807 : AKO does not honour readines probe for pods when Antrea with NPL is used

* AV-209807 : Adding UT to test that AKO now honours readines probe for pods when Antrea with NPL is used

* AV-209807 : Fix for review comments

### CODE REVIEW

**Comment by DixitAakash** [comment]
The FT execution results for npl suite for the source branch is given below :
```
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
| Absolute Path:Test case                                                          |     Results    |       Duration      |  Start Time |   End Time  |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
|      Test Suite Path and Name : avi-dev/test/avitest/functional/ako/test_npl.py  |                |                     |             |             |
| ++++++++++++++++++++++++++++++++++++++++                                         |                |                     |             |             |
| Test_NPL_SNI_DEDICATED.test_npl_create_ingress                                   |     PASSED     |       00:00:21      |   07:32:15  |   07:32:37  |
| Test_NPL_SNI_DEDICATED.test_npl_pod_readiness                                    |     PASSED     |       00:00:21      |   07:32:37  |   07:32:58  |
| Test_NPL_SNI_DEDICATED.test_npl_multiple_ingresses_same_service                  |     PASSED     |       00:00:10      |   07:32:58  |   07:33:08  |
| Test_NPL_SNI_DEDICATED.test_npl_ingress_update_different_service                 |     PASSED     |       00:00:20      |   07:33:08  |   07:33:28  |
| Test_NPL_SNI_DEDICATED.test_npl_lbsvc_creation                                   |     PASSED     |       00:00:15      |   07:33:28  |   07:33:44  |
| Test_NPL_SNI_DEDICATED.test_npl_lbsvc_readiness                                  |     PASSED     |       00:00:50      |   07:33:44  |   07:34:35  |
| Test_NPL_SNI_DEDICATED.test_delete_apps                                          |     PASSED     |       00:00:01      |   07:34:35  |   07:34:37  |
| Test_NPL_EVH_DEDICATED.test_npl_create_ingress                                   |     PASSED     |       00:00:53      |   07:34:37  |   07:35:30  |
| Test_NPL_EVH_DEDICATED.test_npl_pod_readiness                                    |     PASSED     |       00:00:52      |   07:35:30  |   07:36:23  |
| Test_NPL_EVH_DEDICATED.test_npl_multiple_ingresses_same_service                  |     PASSED     |       00:00:25      |   07:36:23  |   07:36:48  |
| Test_NPL_EVH_DEDICATED.test_npl_ingress_update_different_service                 |     PASSED     |       00:00:25      |   07:36:48  |   07:37:14  |
| Test_NPL_EVH_DEDICATED.test_npl_lbsvc_creation                                   |     PASSED     |       00:00:15      |   07:37:14  |   07:37:29  |
| Test_NPL_EVH_DEDICATED.test_npl_lbsvc_readiness                                  |     PASSED     |       00:00:50      |   07:37:29  |   07:38:20  |
| Test_NPL_EVH_DEDICATED.test_delete_apps                                          |     PASSED     |       00:00:01      |   07:38:20  |   07:38:22  |
| Test_NPL_SNI_NON_DEDICATED.test_npl_create_ingress                               |     PASSED     |       00:00:27      |   07:38:22  |   07:38:49  |
| Test_NPL_SNI_NON_DEDICATED.test_npl_pod_readiness                                |     PASSED     |       00:00:26      |   07:38:49  |   07:39:16  |
| Test_NPL_SNI_NON_DEDICATED.test_npl_multiple_ingresses_same_service              |     PASSED     |       00:00:25      |   07:39:16  |   07:39:41  |
| Test_NPL_SNI_NON_DEDICATED.test_npl_ingress_update_different_service             |     PASSED     |       00:00:15      |   07:39:41  |   07:39:57  |
| Test_NPL_SNI_NON_DEDICATED.test_npl_lbsvc_creation                               |     PASSED     |       00:00:15      |   07:39:57  |   07:40:12  |
| Test_NPL_SNI_NON_DEDICATED.test_npl_lbsvc_readiness                              |     PASSED     |       00:00:50      |   07:40:12  |   07:41:03  |
| Test_NPL_SNI_NON_DEDICATED.test_delete_apps                                      |     PASSED     |       00:00:03      |   07:41:03  |   07:41:06  |
| Test_NPL_EVH_NON_DEDICATED.test_npl_create_ingress                               |     PASSED     |       00:00:43      |   07:41:06  |   07:41:50  |
| Test_NPL_EVH_NON_DEDICATED.test_npl_pod_readiness                                |     PASSED     |       00:00:52      |   07:41:50  |   07:42:43  |
| Test_NPL_EVH_NON_DEDICATED.test_npl_multiple_ingresses_same_service              |     PASSED     |       00:00:41      |   07:42:43  |   07:43:24  |
| Test_NPL_EVH_NON_DEDICATED.test_npl_ingress_update_different_service             |     PASSED     |       00:00:25      |   07:43:24  |   07:43:49  |
| Test_NPL_EVH_NON_DEDICATED.test_npl_lbsvc_creation                               |     PASSED     |       00:00:15      |   07:43:49  |   07:44:05  |
| Test_NPL_EVH_NON_DEDICATED.test_npl_lbsvc_readiness                              |     PASSED     |       00:00:50      |   07:44:05  |   07:44:56  |
| Test_NPL_EVH_NON_DEDICATED.test_delete_apps                                      |     PASSED     |       00:00:00      |   07:44:56  |   07:44:56  |
| -------------------------------------------------------------------------------- |  ------------  |     -----------     | ----------- | ----------- |
|                                                                                  |   Passed: 28   |                     |             |             |
|                                                                                  |    Failed:0    |                     |             |             |
|  Total test cases: 28                                                            |    Skipped:0   | Total Time:00:12:41 |             |             |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
```

**Comment by DixitAakash** [comment]
build ako

**Comment by DixitAakash** [comment]
build ako

**Comment by DixitAakash** [comment]
build ako

**Comment by DixitAakash** [comment]
build ako

**Comment by DixitAakash** [comment]
build ako

**Comment by DixitAakash** [comment]
build ako

File: tests/npltests/npl_pod_test.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ const (
 	defaultHostIP   = "10.10.10.10"
 	defaultPodIP    = "192.168.32.10"
 	defaultPodPort  = 80
-	defaultNodePort = 40001
+	defaultNodePort = 61000
```
Comment (pkoshtavmware) [inline tests/npltests/npl_pod_test.go:54]: Should this port be changed back to 4x/5x as 1.10.5 might not have support for ports in 6x range?
cc @akshayhavile

Comment (DixitAakash) [reply tests/npltests/npl_pod_test.go:54]: @pkoshtavmware the npl port range should not be a problem in UT. Also, 61000-62000 and previously 4x/5x  is only the default range and the user can still specify a different port range so port value may not necessarily be in that range.

**Review by pkoshtavmware — APPROVED** [review]
LGTM

**Comment by DixitAakash** [comment]
build ako

**Pull request merged**